### DISCUSSION TITLE
Intel AMT ACM Bare-Metal Activation

### DISCUSSION THREAD

**Discussion by lpn-ch** [Ylianst/MeshCentral#4518] [General]
Hello,
Bare-metal provisioning is announced in this publication:
[https://meshcentral2.blogspot.com/2021/03/meshcentral-intel-amt-security-and.html]([https://meshcentral2.blogspot.com/2021/03/meshcentral-intel-amt-security-and.html](https://meshcentral2.blogspot.com/2021/03/meshcentral-intel-amt-security-and.html)
But I cannot find any information on how to configure this on the MeshCentral. I checked with netstat and cannot find anything listening on port 9971.
Thank you for your help.

**Comment by si458** [comment]
check out the youtube channel for videos which might be helpful https://www.youtube.com/c/MeshCentral/videos

https://www.youtube.com/watch?v=TaKsFEVaMpg
https://www.youtube.com/watch?v=mStyhe-fSC0
https://www.youtube.com/watch?v=mhq0bsWJEOw

a few links to different amt videos which might help

also helpful docs too - https://ylianst.github.io/MeshCentral/intelamt/

**Comment by lpn-ch** [comment]
These videos do not answer my question.
My question is specific to bare-metal provisioning where MeshCentral should be listening on port 9971 for Hello messages.

**Reply by si458** [reply]
i think you have to have windows/linux installed first to ACTIVATE AMT with meshcentral,
once registered with meshcentral you can then control it physically if the machine has an OS or not,
i dont use AMT sadly as all our machines are too old for AMT

**Reply by lpn-ch** [reply]
This is exactly what bare-metal provisioning is NOT about. Bare-metal means not having an OS on the device and the device will announce itself to the bare-metal provisioning server with a Hello message on port 9971.
I understand you don't use AMT, but maybe someone who does can answer this.

**Comment by Ylianst** [comment]
(I am traveling on vacation, with limited email access) 

You can see the configuration in this image:

![image](https://user-images.githubusercontent.com/1319013/189445210-ba1b8e88-a36e-4236-b384-5dea4b5adfa3.png)

The `AmtProvisioningServer` section in the `settings` section of the config.json will enable this feature. MeshCentral will then listen for activation requests, match against your ACM activation certificates and if everything goes well, will activate and add the device to a Intel AMT only device group. No agent or MeshCMD is involved.

This bare-metal activation server is not enabled by default and only makes sense when activating devices on the local network.

**Reply by lpn-ch** [reply]
Hi Ylian,

Thank you for your reply, I did not have the time to try it until now. There has been some progress but it is still not working. I hope you can find the time to help me to get it working. I will give you as much information as possible.

Let me first tell you what I see with my current configuration. I used Wireshark to monitor port 9971. The client PC does send a Hello message to the MeshCentral server, MeshCentral acknowledges the message, but right away the client PC terminates the TCP connection. Only the client PC sent data in the Hello message, the rest was just the SYN/FIN/ACK messages with no data. It seems the client PC was expecting something from MeshCentral that it did not receive so it closed the connection.

I checked the Hello message and it contains the hash of the root certificate from which I created the AMT provisioning certificate. The AMT provisioning certificate was created according to the information from the Intel website:

[https://www.intel.com/content/dam/support/us/en/documents/technologies/remote-configuration-certificate-selection.pdf](url)

Here is my config file:

```
{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "settings": {
    "AmtProvisioningServer": {
      "port": 9971,
      "deviceGroup": "mesh//kk6qJRm29MLFNj6S9kVgZHYWXRsPUZ96LANKIsUgOgOdr98vxqzrSBPIkcXnjzML",
      "newMebxPassword": "XXXXX",
      "trustedFqdn": "HOME.local",
      "ip": "192.168.1.175"
    }
  },
  "domains": {
    "": {
      "title": "MyServer",
      "title2": "Servername",
      "amtAcmActivation": {
        "log": "amtactivation.log",
        "certs": {
          "HomeServer": {
            "certpfx": "Intel-AMT.pfx",
            "certpfxpass": "XXXXX"
          }
        }
      }
    }
  }
}
```
The "Intel-AMT.pfx" file is in the "meshcentral-data"directory.

Here is the device group screenshot from the web UI:

![Screenshot 2022-10-03 103042](https://user-images.githubusercontent.com/25384886/193536628-a8f27534-63ef-462f-9c24-1e62b6c49430.png)

I took the device group Mesh ID from here:

![Screenshot 2022-10-03 103528](https://user-images.githubusercontent.com/25384886/193536912-2a477e3f-97ac-49a0-a8bb-973c11e12072.jpg)

I prepared the client PC with a USB Key file setup. Here is the command:

usbfile -create setup.bin admin NEW_MEBx_PASSWORD -amt -ztc 1 -hash Home-HomeServer-CA.cer sha256 -prov 1 -dns HOME.local -psadd 192.168.1.175 -pspo 9971

Thank you for your help.

Regards,
Louis-Philippe

**Comment by lpn-ch** [comment]
Hi Ylian,
I hope your vacations went well. I think you are back; can you please help me get this working?
Regards,
Louis

**Comment by lpn-ch** [comment]
Hi Ylian,

I finally had to resort to using the old and faithful Intel SCS/RCS to do my bare metal provisioning. I could not wait anymore.

It works, so this means that my internally created provisioning certificate is good, I used the same one I created for MeshCentral, and that my AMT activation in the PC using the setup.bin file is also good. There is something that I probably did not configure correctly in MeshCentral to get it working properly. I did not find any useful log to debug this issue.

I hope you will get around to check my previous post to help me get this working.

Louis

**Comment by srivera42** [comment]
This is the same problem I'm having. 

ACM activation is successful using MESHCMD, i'm using the same FQDN as my fancy CA cert, my option 15 is set to to my FQDN, and my devices are hardwired to my switch.

Cisco Switch , Windows 2022 server, Hybrid mode config.


```
  "settings": {
    "AmtProvisioningServer": {
      "port": 9971,
      "deviceGroup": "mesh//groupnamehash...",
      "newMebxPassword": "xXXXXX1!,
      "trustedFqdn": "mydomain.com",
      "ip": "meshcentralserverip"
    }
  },
```
